docs/src/main/asciidoc/kafka.adoc

```diff
@@ -2269,16 +2271,18 @@ public class FruitConsumer {
 
     @Incoming("fruits")                                     // <1>
     @Transactional                                          // <2>
-    public void persistFruits(Fruit fruit) {                // <3>
-        fruit.persist();                                    // <4>
+    @ActivateRequestContex                                  // <3>
```

Comment:
Should not be required here, as it's automatic as soon as you add @Transactional.

Reply:
> And I just checked the request context is not active.

As Ozan notted, the request context is not active so even if it works when `@Transactional` is added this may not work for all use cases (for ex when multiple persist will be done on the same method).

Reply:
I think, but it may be confirmed by a Narayana expert, when there is no request context, a non-request based transaction context is created.
But I'm not sure of any side effect, for a JPA standpoint a request context may be needed.

Reply:
We don't look for the `@ActivateRequestContext` annotation.

If the narayana JTA CDI extension is enabled then we install CDI interceptors which check for the `@Transactional` annotation and set up the transaction context according to the value of the TxType element of the annotation.